platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -298,7 +303,105 @@ namespace platoon_strategic_ihp
+
+    // Generate a maneuver ID for new maneuver plans (only used for compose lane change maneuver)
+    std::string PlatoonStrategicIHPPlugin::getNewManeuverId() const {
+        /**
+         * @brief Function to generate new maneuver ID. (Use for genreating new lane change maneuver)
+         * 
+         * @return A newly generated uuid as int.
+         */
+
+        static auto gen = boost::uuids::random_generator(); // Initialize uuid generator
+        
+        return boost::lexical_cast<std::string>(gen()); // generate uuid and convert to string
+    }
+
+    // Find target lane indexes based on downtracks for lane change (used for testing in a two-lane map).
+    int PlatoonStrategicIHPPlugin::find_target_lanelet_id(double start_downtrack, double end_downtrack)
+    {
+        /**
+         * @brief Function to find the target lanelet ID for lane change. (Only for UCLA internal test)
+         * 
+         * @param
+         * start_downtrack: The current downtrack distance, in m.
+         * end_downtrack: The target downtrack distance in the target lane, in m.
+         * 
+         * @return
+         * target_lanelet_id: The lanelet id of the target lane.
+         */
+
+        std::vector<lanelet::ConstLanelet> lanelets_in_path = wm_->getLaneletsBetween(start_downtrack, end_downtrack, false, false);
+        std::vector<double> downtracks,crosstracks;
+        std::vector<lanelet::ConstLanelet> lanelets_samelane,lanelets_sidelane;
+        lanelet::ConstLanelet target_lanelet;
+        int target_lanelet_id;
+        if (!lanelets_in_path.empty())
+        {
+            for (int i = 0; i < lanelets_in_path.size(); ++i)
+            {
+                lanelet::BasicPoint2d p1=lanelets_in_path[i].centerline2d().basicLineString().back();
+                double downtrack=wm_->routeTrackPos(p1).downtrack;
+                double crosstrack=wm_->routeTrackPos(p1).crosstrack;
+                if (abs(crosstrack)>1)//1 is a threshold, the assumption for this condition is that there are two lanes on the road
+                {
+                    lanelets_sidelane.push_back(lanelets_in_path[i]);
+                    downtracks.push_back(downtrack);
+                    crosstracks.push_back(crosstrack);
+                }else
+                {
+                    lanelets_samelane.push_back(lanelets_in_path[i]);
+                }
+            }
+            if (!downtracks.empty())
+            {
+                int maxPosition = max_element(downtracks.begin(),downtracks.end()) - downtracks.begin();
+                target_lanelet=lanelets_sidelane[maxPosition];
+                target_lanelet_id=target_lanelet.id();
+            }
+        }
+        return target_lanelet_id;
+    } 
+
+    // Check if the platoon close to the platoon for cut-in join.
+    bool PlatoonStrategicIHPPlugin::isVehicleNextLane(double frontVehicleDtd, double rearVehicleDtd, double frontVehicleCtd)
```

Comment:
The method name doesn't match the stated stated meaning of the return variable.  I suggest renaming the method isVehicleNearPlatoon(), and updating the description of the return value accordingly.